BUG: Fix misleading ValueError in convolve on empty inputs due to variable swapping

Jongwan93 opened the pull request:
#### Reference issue
Fixes #30272

#### What does this implement/fix?
This PR fixes a misleading error message in `numpy.convolve` when the first argument is empty and the second is not.

Previously, an internal optimization swapped the arguments (`a` and `v`) based on their length before performing the empty check. This caused the `ValueError` to incorrectly report that the **second** argument was empty when it was actually the **first** argument that was empty.

This change moves the validation logic before the optimization swap.

#### Additional information
I have added a regression test case in `numpy/_core/tests/test_numeric.py` to ensure the correct error message is raised.

<img width="619" height="46" alt="image" src="https://github.com/user-attachments/assets/f99dd8f6-3182-434b-a41b-7f23e8a634ea" />

Jongwan93 committed:
BUG: Fix misleading ValueError in convolve on empty inputs
Previously, numpy.convolve swapped arguments for optimization if the
second argument was longer than the first. This occurred before
checking for empty arrays. Consequently, if the first argument was
empty, the swap would cause the error message to incorrectly report
that the second argument was empty (e.g., "v cannot be empty").

This commit moves the validation of empty inputs to occur before the
optimization swap. This ensures that the ValueError correctly
identifies which argument is actually empty.

Jongwan93 committed:
TST: Add regression test for convolve empty input error message
This adds a regression test to ensure `numpy.convolve` raises the
correct ValueError when the first argument is empty. Previously,
internal variable swapping caused the error message to incorrectly
attribute the empty input to the wrong argument.

Jongwan93 committed:
minor fix: variable name change

seberg reviewed (commented):
I like that the fix is to just move things, but the test is failing, mixing up the parameter names.
(You wrote the test as if the variable name matters, but the argument name matters and we are passing things positionally.)

Jongwan93 committed:
TST: Simplify regression test
Refactored the regression test for `convolve` to directly use `np.array`
instead of named variables, avoiding confusion about parameter names vs
argument names. Also removed unnecessary brakets in numeric.py

Jongwan93 commented:
@seberg 
Thanks for the review!

I've refactored the test to pass arrays directly  to avoid confusion between variable names and argument positions.
I also reverted the unrelated formatting changes.

Let me know if it looks okay now!

Jongwan93 committed:
Minor fix: missing new line added at the end of test_numeric.py file

Jongwan93 commented:
> Lint error needs fixing, just needs the file to end with a newline.

Just added missing new line at the end of `test_numeric.py` file. Thank you

WarrenWeckesser committed:
Update numpy/_core/tests/test_numeric.py
